### PR TITLE
Fix apidoc routes with base url

### PR DESCRIPTION
**Pull request by crobibero**
Removes the baseUrl from the openapi endpoints if a baseurl is set.
#4029

image

### CODE REVIEW

**Comment by cvium** [comment]
I don't know if I like this change. Now that we're no longer using ServiceStack, we can actually use this middleware to set the base url https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.builder.usepathbaseextensions.usepathbase?view=aspnetcore-3.1.

I think Swashbuckle will fix itself, but it does require a few more changes to use PathBase because we rely on `Request.Path` in some places and it does not contain `PathBase`. Simple to fix though.

**Comment by cvium** [comment]
`UsePathBase` seems intended for reverse proxies stripping the base url. I can't find much information about global route prefixing that isn't using the `IApplicationModelConvention`.

I am still not sure which of the two solutions is better.

**Comment by cromefire** [comment]
I'm by no means an ASP expert (I'm probably not even a good ASP developer) but the middleware sounds like a lot cleaner solution as this would essentially remove base url, from a lot of places where it's not needed, right?

**Comment by cvium** [comment]
Yes in a way. It matches the segment against the Path and moves the match from Path to PathBase so you end up with `Path=/mycontroller` and `PathBase=/baseurl`.

**Comment by crobibero** [comment]
Another issue that cvium brought up is that with this change the endpoints will listen to both `/` and `/baseurl/`

**Comment by crobibero** [comment]
https://github.com/aspnet/HttpAbstractions/issues/893#issuecomment-313911174

**Comment by crobibero** [comment]
What this PR does:

maps all routes to the base url so they only listen on the base url.
-Allows openapi spec to generate using the correct server & base url.

Change base url rewrite to actually work.
Host web content using `.UseStaticFiles` instead of a controller.

File: Jellyfin.Server/Extensions/ApiServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -151,7 +150,6 @@ public static IMvcBuilder AddJellyfinApi(this IServiceCollection serviceCollecti
                 })
                 .AddMvc(opts =>
                 {
-                    opts.UseGeneralRoutePrefix(baseUrl);
```
Comment (cvium) [inline Jellyfin.Server/Extensions/ApiServiceCollectionExtensions.cs:154]: If the iapplicationmodel is no longer used, then please delete it.

**Review by Ullmie02 — APPROVED** [review]

**Review by cvium — APPROVED** [review]

**Comment by cromefire** [comment]
Yay 🎉

**Pull request merged**